pkg/pipeline/instance.go

```diff
@@ -58,10 +58,20 @@ type Instance struct {
 	ConnectorIDs []string
 	ProcessorIDs []string
 
+	status     Status
+	statusLock sync.RWMutex
+
 	n map[string]stream.Node
 	t *tomb.Tomb
 }
 
+// encodableInstance is an encodable "view" of Instance
+// through which we can also encode an Instance's unexported fields.
+type encodableInstance struct {
+	*Instance
+	Status Status
```

Comment:
Do we want `Status` to be unexported here as well to ensure we're not mutating an encoded status?